soca travis CI broken (in ufo)

## Description

https://travis-ci.com/github/JCSDA-internal/soca/builds/216670131

> -- Configuring done
CMake Error at /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:294 (add_library):
  Target "ufo" links to target "gsl::gsl-lite" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:639 (ecbuild_add_library_impl)
  src/ufo/CMakeLists.txt:125 (ecbuild_add_library)

This is likely because we started using ioda-engines in ufo. There's probably something missing in ufo CMakeLists; @travissluka, @rhoneyager, @srherbener  could you take a look?

Now, with [this ufo branch](https://github.com/JCSDA-internal/ufo/compare/bugfix/soca-travisci), it [fails with a different error](https://travis-ci.com/github/JCSDA-internal/soca/builds/216791576):
> -- Configuring done
CMake Error at /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:294 (add_library):
  Target "ufo" links to target "MPI::MPI_C" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:639 (ecbuild_add_library_impl)
  src/ufo/CMakeLists.txt:125 (ecbuild_add_library)
CMake Error at /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:294 (add_library):
  Target "ufo" links to target "MPI::MPI_C" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  /usr/local/share/ecbuild/cmake/ecbuild_add_library.cmake:639 (ecbuild_add_library_impl)
  src/ufo/CMakeLists.txt:125 (ecbuild_add_library)


I have no idea how to fix it, unfortunately.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,7 @@ include_directories( ${SABER_INCLUDE_DIRS} )
 # ioda
 ecbuild_use_package( PROJECT ioda VERSION 0.1.0 REQUIRED )
 include_directories( ${IODA_INCLUDE_DIRS} )
+find_package(gsl-lite REQUIRED HINTS $ENV{gsl_lite_DIR})
 
 # ufo
 ecbuild_use_package( PROJECT ufo VERSION 0.1.0 REQUIRED )

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ ecbuild_use_package( PROJECT mom6 REQUIRED )
 include_directories( ${MOM6_INCLUDE_DIRS} )
 
 # MPI
-ecbuild_find_mpi( COMPONENTS CXX Fortran REQUIRED )
+ecbuild_find_mpi( COMPONENTS C CXX Fortran REQUIRED )
 ecbuild_include_mpi()
 link_libraries(${MPI_CXX_LIBRARIES} ${MPI_Fortran_LIBRARIES})
 

